include/odp/api/spec/classification.h

```diff
@@ -644,6 +689,24 @@ void odp_cls_pmr_param_init(odp_pmr_param_t *param);
 odp_pmr_t odp_cls_pmr_create(const odp_pmr_param_t *terms, int num_terms,
 			     odp_cos_t src_cos, odp_cos_t dst_cos);
 
+/**
+ * Create a packet matching rule with options
+ *
+ * Similar to odp_cls_pmr_create() function with additional PMR creation
+ * options specified through odp_pmr_create_opt_t.
+ *
+ * Use odp_cls_pmr_create_opt_init() to initialize options into their default
+ * values.
+ *
+ * @param opt	       points to PMR create options
+ * @param src_cos      source CoS handle
+ * @param dst_cos      destination CoS handle
+ *
+ * @return Handle to the Packet Match Rule.
+ * @retval ODP_PMR_INVALID on failure
+ */
+odp_pmr_t odp_cls_pmr_create_opt(const odp_pmr_create_opt_t *opt,
+				 odp_cos_t src_cos, odp_cos_t dst_cos);
```

Comment:
Need to document if old pmr_create and new pmr_create_opt calls can be mixed. I think those could be mixed, just need to document what happens to mark when PMR created with the old function hits/misses. Is the current  mark value set to zero, or not changed.

Reply:
Yes. We can document the old and new API can be mixed. If we use old API then odp_packet_mark() shall return ODP_CLS_MARK_DEFAULT;

Reply:
Added the following note in the next version.
 * @note Application can mix the usage of odp_cls_pmr_create() and
 * odp_cls_pmr_create_opt(), when mixed and the PMR is created with
 * odp_cls_pmr_create() then it will be similar to PMR creation using
 * odp_cls_pmr_create_opt() with default options.